docs/directory-structure.md

```diff
@@ -1,241 +1,213 @@
+
+    There are advantages and disadvantages to having a single django
+    process serving multiple sites, or having multiple sites share a single
+    database; however, this proposed compartmentalization should allow
+    reduced risk when experimenting with new languages.
+
+  - For now, all the code for all the dictionary applications will reside
+    in this git repo, and not be split into separate git repos.
+
+    That will make it much easier to move code around during development,
+    and to run tests across all dictionary applications when making
+    changes.
+
+    Some day, we hope, when mature and stable, morphodict could go on PyPI
+    as its own framework package with instructions on how to set up new
+    languages without being in this git repo. We are very far from that
+    point right now, but it’s something to keep in mind as a long-term
+    goal.
+
+[639-3]: https://iso639-3.sil.org/code_tables/639/data
+
+### Proposed layout
+
+    $repo
+    ├── .git/
+    ├── package.json                # dependencies for bundlers, JS/CSS frameworks
+    ├── Pipfile
+    ├── arpeng-manage.py            # manage.py scripts are at the top-level for easy access
+    ├── crkeng-manage.py
+    ├── crkfra-manage.py
+    ├── cwdeng-manage.py
+    ├── srseng-manage.py
+    └── src/
+        ├── CreeDictionary/         # existing code, eventually goes away
+        │   ├── __init__.py
+        │   ├── API/                # this name goes away :(
+        │   └── CreeDictionary/     # this goes away too
+        │       ├── __init__.py
+        │       ├── models.py
+        │       ├── views.py
+        │       └── …
+        │
+        ├── morphodict/             # python package for language-independent code
+        │   ├── __init__.py
+        │   ├── cvd/
+        │   ├── lexicon/            # django app with primary database tables
+        │   │   ├── __init__.py
+        │   │   ├── models.py
+        │   │   ├── management/
+        │   │   │   └── commands/
+        │   │   │       └── dictimport.py
+        │   │   ├── parser.py
+        │   │   ├── test_parser.py  # test_* files are mixed in with non-test source code
+        │   │   └── testdata/       # Use `testdata` directories for test data
+        │   ├── paradigm_filler/
+        │   ├── frontend/
+        │   │   ├── foo.js
+        │   │   ├── foo.css
+        │   ⋮   └── …
+        │
+        ├── crkeng/                 # python package for itwêwina
+        │   ├── __init__.py
+        │   ├── cypress/
+        │   │   └── …
+        │   ├── docker/
+        │   │   └── …
+        │   ├── res/                # Resources go here
+        │   │   ├── altlabels.tsv
+        │   │   ├── dictionaries/
+        │   │   ├── fst/
+        │   │   └── layouts/
+        │   ├── site/               # Django project
+        │   │   ├── __init__.py
+        │   │   ├── settings.py
+        │   │   └── urls.py
+        │   └── frontend/           # Not a python package
```

Comment:
Some more structure here would be helpful, depending on how complex the frontend is. Is there an expectation that the entire frontend will only have one JS and one CSS asset per project? If so, these should probably be given a naming convention of `app.js` and `app.css` or some such. Or are we using a component structure, like below?

```
frontend/
└── component
    ├── component.css
    ├── component.js
    └── component.test.js
```

Reply:
I’ve updated the docs with a bit more info on this. In short:

  - There already is a bit of structure to our frontend stuff, with multiple JS and CSS files
  - That seems to be working, so we’ll keep it for now
  - Although language-specific stuff should move to language-specific directories